src/util/namespace.h

```diff
@@ -33,8 +33,7 @@ class namespace_baset
   const symbolt &lookup(const irep_idt &name) const
   {
     const symbolt *symbol;
-    if(lookup(name, symbol))
-      throw "identifier "+id2string(name)+" not found";
+    CHECK_RETURN(!lookup(name, symbol));
```

Comment:
lookup is side-effectful so shouldn't be in an INVARIANT

What to use here is a bit tricky again, because technically it's a precondition that the name exists in the namespace but that doesn't fit the definition of PRECONDITION, so I'd just do something like

```
const symbolt *symbol;
bool not_found = lookup(name, symbol); // this is really counterintuitive X_X
INVARIANT(!not_found, "We are assuming that a name exists in the namespace when this function is called");
return *symbol;
```